File: src/dataset/Java/bob_exercism-johannesrave-education/claude/ClaudeSonnet4_Bob_v3.java
import java.util.regex.Pattern;

class Bob{
    private static final Pattern SHOUTING_PATTERN = Pattern.compile("([^a-z]*[A-Z][A-Z]+[^a-z]*)");
    private static final Pattern QUESTION_PATTERN = Pattern.compile(".*\\?\\s*$");
    private static final Pattern SILENCE_PATTERN = Pattern.compile("^\\s*$");
    
    public Bob(){
    }
    
    public String hey(String phrase){
        boolean isShouting = SHOUTING_PATTERN.matcher(phrase).matches();
        boolean isQuestion = QUESTION_PATTERN.matcher(phrase).matches();
        boolean isSilence = SILENCE_PATTERN.matcher(phrase).matches();
        
        if (isShouting && isQuestion){
            return "Calm down, I know what I'm doing!";
        } else if (isQuestion){
            return "Sure.";
        } else if (isShouting){
            return "Whoa, chill out!";
        } else if (isSilence){
            return "Fine. Be that way!";
        } else {
            return "Whatever.";
        }
    }
}

File: src/dataset/cpp/bob_exercism-tqa236/claude/ClaudeSonnet4_bob_v3.cpp
#include "bob.h"

std::string bob::hey(std::string text) {
  bool has_uppercase = false;
  bool has_lowercase = false;
  bool is_question = false;
  bool has_non_space = false;
  
  for (const char c : text) {
    if (!std::isspace(c)) {
      has_non_space = true;
      if (c == '?') {
        is_question = true;
      }
    }
    if (std::islower(c)) {
      has_lowercase = true;
    } else if (std::isupper(c)) {
      has_uppercase = true;
    }
    
    if (has_uppercase && has_lowercase && has_non_space && is_question) {
      break;
    }
  }
  
  if (!has_non_space) {
    return "Fine. Be that way!";
  }
  
  const bool is_yelling = has_uppercase && !has_lowercase;
  
  if (is_yelling && is_question) {
    return "Calm down, I know what I'm doing!";
  }
  if (is_yelling) {
    return "Whoa, chill out!";
  }
  if (is_question) {
    return "Sure.";
  }
  return "Whatever.";
}